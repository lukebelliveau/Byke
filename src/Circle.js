import React from 'react';
import { View } from 'react-native';

export default ({ color }) => <View style={style(color)} />;

const style = color => ({
  width: 15,
  height: 15,
  borderRadius: 20 / 2,
  backgroundColor: color,
});
