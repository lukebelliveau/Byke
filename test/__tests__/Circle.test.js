import React from 'react';
import renderer from 'react-test-renderer';
import Circle from '../../src/components/shared/Circle';

describe('Marker', () => {
  it('should be green', () => {
    const marker = renderer.create(<Circle color={'green'} />).toJSON();

    expect(marker).toMatchSnapshot();
  });

  it('should be red', () => {
    const marker = renderer.create(<Circle color={'red'} />).toJSON();

    expect(marker).toMatchSnapshot();
  });
});
