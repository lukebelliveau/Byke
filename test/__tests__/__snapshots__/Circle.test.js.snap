// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Marker should be green 1`] = `
<View
  style={
    Object {
      "backgroundColor": "green",
      "borderRadius": 10,
      "height": 15,
      "width": 15,
    }
  }
/>
`;

exports[`Marker should be red 1`] = `
<View
  style={
    Object {
      "backgroundColor": "red",
      "borderRadius": 10,
      "height": 15,
      "width": 15,
    }
  }
/>
`;
